kong/plugins/hmac-auth/access.lua

```diff
@@ -196,26 +262,33 @@ local function do_authentication(conf)
 
   -- retrieve hmac parameter from Proxy-Authorization header
   local hmac_params = retrieve_hmac_fields(ngx.req, headers, PROXY_AUTHORIZATION, conf)
+
   -- Try with the authorization header
   if not hmac_params.username then
     hmac_params = retrieve_hmac_fields(ngx.req, headers, AUTHORIZATION, conf)
   end
-  if not (hmac_params.username and hmac_params.signature) then
+
+  local ok, err = validate_params(hmac_params, conf)
+  if not ok then
+    ngx_log(ngx.DEBUG, err)
     return false, {status = 403, message = SIGNATURE_NOT_VALID}
   end
 
   -- validate signature
   local credential = load_credential(hmac_params.username)
   if not credential then
+    ngx_log(ngx.DEBUG, "failed to retrieve credential for " .. hmac_params.username)
```

Comment:
We should avoid string concatenation operators and use variadic arguments when possible. Here: 
```lua
ngx_log(ngx.DEBUG, "failed to retrieve credential for ", hmac_params.username)
```